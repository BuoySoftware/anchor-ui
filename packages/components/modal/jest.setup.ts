import "@testing-library/jest-dom";
import "jest-styled-components";

jest.mock("react-modal", () => {
  const actual = jest.requireActual("react-modal");

  return {
    ...actual,
    setAppElement: (args: string) => {
      try {
        actual.setAppElement(args);
      } catch {
        console.warn(args, "Is not in DOM");
      }
    },
  };
});
// jest.spyOn(ReactModal, "setAppElement").mockImplementation(() => undefined);
