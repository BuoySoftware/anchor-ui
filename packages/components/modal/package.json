{
  "name": "@buoysoftware/anchor-modal",
  "version": "0.33.4",
  "main": "src/index.ts",
  "license": "MIT",
  "dependencies": {
    "@buoysoftware/anchor-layout": "^0.31.1",
    "@buoysoftware/anchor-theme": "^0.31.1",
    "@buoysoftware/anchor-typography": "^0.31.2",
    "@types/react-modal": "^3.13.1",
    "polished": "^4.2.2",
    "react-modal": "^3.16.1",
    "styled-components": "^5.3.8"
  },
  "scripts": {
    "build": "tsc -outDir dist",
    "compile": "tsc --noEmit",
    "postpack": "clean-package restore",
    "prepack": "clean-package",
    "test": "jest"
  },
  "devDependencies": {
    "jest": "^29.3.1",
    "ts-jest": "^29.0.4",
    "ts-node": "^10.9.1"
  }
}
