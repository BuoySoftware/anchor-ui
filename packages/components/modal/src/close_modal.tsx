import { Box } from "@buoysoftware/anchor-layout";

interface OwnProps {
  closeModal?: (e: React.MouseEvent | React.KeyboardEvent) => void;
}

type CloseModalProps = OwnProps;

export const CloseModal: React.FC<CloseModalProps> = ({
  closeModal,
}): React.ReactElement => (
  <Box
    aria-label="Close Button"
    height="26px"
    onClick={closeModal}
    position="relative"
    role="button"
    style={{ cursor: "pointer" }}
    width="26px"
  >
    <Box
      bg="grey50"
      height="1.5px"
      left="5.5px"
      position="absolute"
      style={{ rotate: "45deg" }}
      top="50%"
      width="15px"
    />
    <Box
      bg="grey50"
      height="1.5px"
      left="5.5px"
      position="absolute"
      style={{ rotate: "135deg" }}
      top="50%"
      width="15px"
    />
  </Box>
);
