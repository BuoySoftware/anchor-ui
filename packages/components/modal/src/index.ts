export * from "./modal";
export { useModal } from "./useModal";
