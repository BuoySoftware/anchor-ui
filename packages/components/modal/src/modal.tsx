import { Heading } from "@buoysoftware/anchor-typography";
import { Box } from "@buoysoftware/anchor-layout";
import ReactModal, { Props as ReactModalProps } from "react-modal";
import { rgba } from "polished";
import { useTheme } from "@buoysoftware/anchor-theme";
import { get } from "styled-system";

ReactModal.setAppElement("#root");

interface OwnProps {
  bg?: string;
  borderRadius?: string;
  children?: React.ReactNode;
  modalActions?: React.ReactNode;
  testId: string;
  title: string;
  width?: string;
}

export type ModalProps = ReactModalProps & OwnProps;

export const Modal: React.FC<ModalProps> = ({
  bg = "white",
  borderRadius = "modal",
  children,
  modalActions,
  testId,
  title,
  width = "modalWidth",
  ...modalProps
}): React.ReactElement => {
  const { colors, radii, sizes, space } = useTheme();

  const contentWidth = get(sizes, width, width);
  const contentRadius = get(radii, borderRadius, borderRadius);
  const contentBackground = get(colors, bg, bg);

  return (
    <ReactModal
      style={{
        overlay: {
          backgroundColor: rgba(colors.blue100, 0.64),
          display: "flex",
          alignItems: "flex-start",
          justifyContent: "center",
          zIndex: 99,
        },
        content: {
          backgroundColor: contentBackground,
          border: "none",
          borderRadius: contentRadius,
          bottom: undefined,
          left: undefined,
          marginTop: space.modal.gutterT,
          padding: 0,
          position: undefined,
          right: undefined,
          top: undefined,
          width: contentWidth,
        },
      }}
      {...modalProps}
    >
      <Box data-testid={testId} py="xl" px="modal.gutterX">
        <Box mb="xl">
          <Heading data-testid={`${testId}-title`} size="xl">
            {title}
          </Heading>
        </Box>
        <Box maxHeight="60vh" overflow="none" mb="l">
          {children}
        </Box>
        <Box display="flex" justifyContent="flex-end" mt="l">
          {modalActions}
        </Box>
      </Box>
    </ReactModal>
  );
};
