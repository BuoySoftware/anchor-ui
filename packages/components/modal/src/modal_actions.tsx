import { Box, BoxProps } from "@buoysoftware/anchor-layout";
import styled from "styled-components";

interface OwnProps {
  children?: React.ReactNode;
}

export type ModalActionsProps = BoxProps & OwnProps;

const ModalActions = styled(({ children, ...boxProps }) => (
  <Box
    borderTop="1SolidGray"
    display="flex"
    justifyContent="flex-end"
    pt="l"
    mx="modal.gutterX"
    pb="xxl"
    {...boxProps}
  >
    {children}
  </Box>
))<ModalActionsProps>``;

export default ModalActions;
