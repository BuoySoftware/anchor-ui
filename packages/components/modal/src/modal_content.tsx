import { Box } from "@buoysoftware/anchor-layout";

interface OwnProps {
  children?: React.ReactNode;
}

type ModalContentProps = OwnProps;

const ModalContent: React.FC<ModalContentProps> = ({
  children,
}): React.ReactElement => {
  return (
    <Box maxHeight="60vh" overflow="auto" py="xxl" px="modal.gutterX">
      {children}
    </Box>
  );
};

export default ModalContent;
