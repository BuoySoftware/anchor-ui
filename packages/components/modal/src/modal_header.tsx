import { Box } from "@buoysoftware/anchor-layout";
import { Body } from "@buoysoftware/anchor-typography";

interface OwnProps {
  title: string;
}

export type ModalHeaderProps = OwnProps;

const ModalHeader: React.FC<ModalHeaderProps> = ({
  title,
}): React.ReactElement => {
  return (
    <Box borderBottom="1SolidGray" py="xl" px="xxxl">
      <Body data-testid="modal-title" variant="subtitle">
        {title}
      </Body>
    </Box>
  );
};

export default ModalHeader;
