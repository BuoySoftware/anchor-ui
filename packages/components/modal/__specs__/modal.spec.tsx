import { Button } from "@buoysoftware/anchor-button";

import { Modal } from "../src/modal";
import { render } from "../test_utils";

const MockModal: React.FC = (): React.ReactElement => {
  return (
    <Modal
      isOpen={true}
      modalActions={
        <>
          <Button onClick={jest.fn()} colorScheme="basic" mr="xxs">
            Cancel
          </Button>
          <Button onClick={jest.fn()} colorScheme="primary">
            Save
          </Button>
        </>
      }
      testId="test-modal"
      title="Test Modal"
    >
      Test modal contents
    </Modal>
  );
};

describe("<Modal />", () => {
  it("renders correctly", () => {
    const modal = render(<MockModal />);

    expect(modal.baseElement).toMatchSnapshot();
  });
});
