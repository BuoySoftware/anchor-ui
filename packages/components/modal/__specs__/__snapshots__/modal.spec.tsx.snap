// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
.c9 {
  box-sizing: border-box;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 1.25rem;
  padding-right: 1.25rem;
  margin-right: 0.25rem;
}

.c12 {
  box-sizing: border-box;
  color: inherit;
  width: s;
  height: s;
}

.c14 {
  box-sizing: border-box;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 1.25rem;
  padding-right: 1.25rem;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  cursor: pointer;
  border-style: solid;
  border-width: 1px;
  background-color: #FFFFFF;
  border-color: #7F8387;
  color: #101112;
}

.c11:hover {
  background-color: #F5F6F7;
}

.c11:active {
  background-color: #D9DDE1;
}

.c11:disabled {
  cursor: not-allowed;
  background-color: #FFFFFF;
  border-color: #BABEC3;
  color: #9CA1A5;
}

.c11:focus-visible {
  border-radius: 4px;
  outline: 2px solid #5b91f4;
  outline-offset: 2px;
}

.c15 {
  cursor: pointer;
  border-style: solid;
  border-width: 1px;
  background-color: #276EF1;
  border-color: transparent;
  color: #FFFFFF;
}

.c15:hover {
  background-color: #1E54B7;
}

.c15:active {
  background-color: #174291;
}

.c15:disabled {
  cursor: not-allowed;
  background-color: #EBEDF0;
  border-color: transparent;
  color: #9CA1A5;
}

.c15:focus-visible {
  border-radius: 4px;
  outline: 2px solid #5b91f4;
  outline-offset: 2px;
}

.c13 {
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Segoe UI, Roboto,Helvetica Neue,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  box-sizing: border-box;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  padding-left: 2rem;
  padding-right: 2rem;
}

.c1 {
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.5rem;
}

.c3 {
  box-sizing: border-box;
  color: #101112;
  width: xl;
  height: xl;
}

.c5 {
  box-sizing: border-box;
  height: 26px;
  width: 26px;
  position: relative;
}

.c6 {
  box-sizing: border-box;
  background-color: #9CA1A5;
  height: 1.5px;
  width: 15px;
  left: 5.5px;
  position: absolute;
  top: 50%;
}

.c7 {
  box-sizing: border-box;
  max-height: 60vh;
  overflow: none;
  margin-bottom: 1.25rem;
}

.c8 {
  box-sizing: border-box;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1.25rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Segoe UI, Roboto,Helvetica Neue,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 1.5rem;
  font-weight: 600;
  line-height: 1.75rem;
}

<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(10, 25, 53, 0.64); display: flex; align-items: center; justify-content: center; z-index: 99;"
    >
      <div
        aria-modal="true"
        class="ReactModal__Content"
        role="dialog"
        style="background: rgb(255, 255, 255); overflow: auto; border-radius: 30px; outline: none; padding: 0px; background-color: rgb(255, 255, 255); margin-top: 0px; width: 527px;"
        tabindex="-1"
      >
        <div
          class="c0"
          data-testid="test-modal"
        >
          <div
            class="c1 c2"
            display="flex"
          >
            <div
              class="c3 c4"
              color="text.primary"
              data-testid="test-modal-title"
              font-size="font-size-400"
              font-weight="600"
            >
              Test Modal
            </div>
            <div
              aria-label="Close Button"
              class="c5"
              height="26px"
              role="button"
              style="cursor: pointer;"
              width="26px"
            >
              <div
                class="c6"
                height="1.5px"
                width="15px"
              />
              <div
                class="c6"
                height="1.5px"
                width="15px"
              />
            </div>
          </div>
          <div
            class="c7"
            data-testid="test-modal-body"
            overflow="none"
          >
            Test modal contents
          </div>
          <div
            class="c8"
            data-testid="test-modal-actions"
            display="flex"
          >
            <button
              class="c9 c10 c11"
              display="flex"
              height="40"
              type="button"
            >
              <span
                class="c12 c13"
                color="inherit"
                font-size="font-size-100"
                font-weight="600"
                text-decoration="none"
              >
                Cancel
              </span>
            </button>
            <button
              class="c14 c10 c15"
              display="flex"
              height="40"
              type="button"
            >
              <span
                class="c12 c13"
                color="inherit"
                font-size="font-size-100"
                font-weight="600"
                text-decoration="none"
              >
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
